### PR TITLE
Remove unnecessary usages of toArray().

### PR DESCRIPTION
**Pull request by LucasSloan**
We were using toArray to use iterators in for of loops.  Since we've set the flag that has TypeScript downlevel them for us, we don't need to do it manually, so they're redundant.

### CODE REVIEW

**Comment by LucasSloan** [comment]
Unfortunately, no.  We also use it when something expects an array, but we want to provide an iterator (ie we want to pass in the keys of a map as an array).

**Comment by evmar** [comment]
Can we use Array.from() there?

**Comment by LucasSloan** [comment]
Yes we can, thanks!

File: test/test_support.ts [hunk 1]
```diff
@@ -253,13 +253,12 @@ export function extractInlineSourceMap(source: string): BasicSourceMapConsumer {
...
 
 export function findFileContentsByName(filename: string, files: Map<string, string>): string {
-  const filePaths = toArray(files.keys());
-  for (const filepath of filePaths) {
+  for (const filepath of files.keys()) {
     if (path.parse(filepath).base === path.parse(filename).base) {
       return files.get(filepath)!;
     }
   }
-  assert(undefined, `Couldn't find file ${filename} in files: ${JSON.stringify(filePaths)}`);
```
Comment (evmar) [inline test/test_support.ts]: I think the reason for the JSON.stringify here is that it quotes the elements of the list.  I think we want to keep that.

files.toString()
"a,b.ts"
JSON.stringify(files)
"["a","b.ts"]"

File: src/type-translator.ts [hunk 2]
```diff
@@ -532,7 +531,8 @@ export class TypeTranslator {
       return `function(new: (${constructedType})${paramsStr}): ?`;
     }
 
-    for (const field of toArray(type.symbol.members.keys())) {
+    // tslint:disable-next-line:no-any
+    for (const field of (type.symbol.members.keys() as any)) {
```
Comment (evmar) [inline src/type-translator.ts]: Why is this needed?  Isn't members already a Map?

Comment (LucasSloan) [reply src/type-translator.ts]: It's a ts.Map, which wants to be a real Map, but isn't for some reason.

File: test/test_support.ts (hunk 1 above)
Comment (LucasSloan) [reply test/test_support.ts]: Done.

File: src/type-translator.ts (hunk 2 above)
Comment (evmar) [reply src/type-translator.ts]: It looks like it's just named Map, but it's not actually compatible with ES6.  I would guess this code doesn't work as written.

file: 'file:///usr/local/google/home/evanm/projects/tsickle/src/type-translator.ts'
severity: 'Error'
message: 'Type 'Iterator<string>' is not an array type or a string type or does not have a '[Symbol.iterator]()' method that returns an iterator.'
at: '535,21'
source: 'ts'

Comment (LucasSloan) [reply src/type-translator.ts]: The comments on the type say it's an ES6 map.  I'm pretty sure that the object is a proper map, they just produced a bad .d.ts for some reason.  All the tests pass, which wouldn't be the case if this threw an error instead of doing the right thing.

Comment (evmar) [reply src/type-translator.ts]: Can you add a comment explaining this?

Comment (LucasSloan) [reply src/type-translator.ts]: Done.
